File: src/backend_java/src/main/java/com/group07/human_resource_management/entity/PointConversionRule.java
package com.group07.human_resource_management.entity;

import jakarta.persistence.*;
import lombok.*;
import java.time.LocalDateTime;

@Entity
@Table(name = "point_conversion_rules")
@Getter @Setter
@Builder @NoArgsConstructor @AllArgsConstructor
public class PointConversionRule {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "point_value", nullable = false)
    private Integer pointValue;

    @Column(name = "money_value", nullable = false)
    private Double moneyValue;

    // updated_by → employee.id
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "updated_by")
    private Employee updatedBy;

    @Column(name = "updated_at")
    private LocalDateTime updatedAt;

    @Column(name = "is_active")
    private Boolean isActive;
}

File: src/backend_dotnet/HRMApi/Models/PointConversionRule.cs
using System;
using System.Collections.Generic;

namespace HRMApi.Models;

public partial class PointConversionRule
{
    public int Id { get; set; }

    public int PointValue { get; set; }

    public decimal MoneyValue { get; set; }

    public int? UpdatedBy { get; set; }

    public DateTime? UpdatedAt { get; set; }

    public bool? IsActive { get; set; }

    public virtual Employee? UpdatedByNavigation { get; set; }
}
